### PR TITLE
Deprecate non generic filtering

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Add deprecations to non generic filtering 

This PR adds deprecation to attribute relationship filters using the suffix syntax such as `_EQ` / `_SOME`

### CODE REVIEW

File: packages/graphql/src/schema/generation/augment-where-input.ts [hunk 1]
```diff
@@ -250,9 +269,15 @@ function fieldConfigsToFieldConfigMap({
...
+        let directives;
+        if (deprecatedDirectives.length) {
+            directives = deprecatedDirectives;
+        } else if (field.deprecationReason) {
+            directives = [{ name: DEPRECATED, args: { reason: field.deprecationReason } }];
+        }
         fieldsConfigMap[field.name] = {
             type: field.typeName,
-            directives: deprecatedDirectives,
+            directives,
```
Comment (mjfwebb) [inline packages/graphql/src/schema/generation/augment-where-input.ts:280]: Probably fine, but what happens if directives is undefined?

File: .changeset/nine-games-clap.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+"@neo4j/graphql": patch
+---
+
+
```
Comment (mjfwebb) [inline .changeset/nine-games-clap.md]:
```suggestion

```

File: .changeset/nine-games-clap.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+Deprecate relationship filtering using the non-generic version such as `actors_SOME:{ title_EQ: "The Matrix" }` in favor of the dedicated input `actors: { some: { title: { eq: "The Matrix" } } }`.
```
Comment (mjfwebb) [inline .changeset/nine-games-clap.md]:
```suggestion
Deprecate relationship filtering using the non-generic version such as `actors_SOME: { title_EQ: "The Matrix" }` in favor of the dedicated input `actors: { some: { title: { eq: "The Matrix" } } }`.
```

Comment (mjfwebb) [reply .changeset/nine-games-clap.md]: I see the terms "non-generic" and "dedicated" both used here. Is that intentional?

**Review by mjfwebb — COMMENTED** [review]
Nice! Just a couple of comments.

File: .changeset/nine-games-clap.md (hunk 3 above)
Comment (MacondoExpress) [reply .changeset/nine-games-clap.md]: Good point, not there was no specific reasoning behind!

File: packages/graphql/src/schema/generation/augment-where-input.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/src/schema/generation/augment-where-input.ts:280]: From the typings it should not be possible!

**Review by mjfwebb — APPROVED** [review]

**Pull request merged**